locust/runners.py

```diff
@@ -499,8 +514,6 @@ def client_listener(self):
             elif msg.type == "exception":
                 self.log_exception(msg.node_id, msg.data["msg"], msg.data["traceback"])
 
-            if not self.state == STATE_INIT and all(map(lambda x: x.state != STATE_RUNNING and x.state != STATE_HATCHING, self.clients.all)):
-                self.state = STATE_STOPPED
```

Comment:
I think we still want this check here so that we'll change the state to STOPPED immediately when we've received the stop message from the last worker (under normal operation where workers don't go missing).

Reply:
I made a new function for this and call it both here and in heartbeat_worker